plugin-server/src/cdp/consumers/cdp-behavioural-events.consumer.test.ts

```diff
@@ -196,4 +204,267 @@ describe('CdpBehaviouralEventsConsumer', () => {
+                'SELECT count FROM behavioral_event_counters WHERE team_id = ? AND filter_hash = ? AND person_id = ? AND date = ?',
+                [team.id, filterHash, CassandraTypes.Uuid.fromString(personId), today],
+                { prepare: true }
+            )
+
+            expect(cassandraResult.rows).toHaveLength(1)
+            expect(cassandraResult.rows[0].count.toNumber()).toBe(2)
+        })
+
+        it('should not write counter when event does not match', async () => {
+            // Arrange
+            const personId = '550e8400-e29b-41d4-a716-446655440000'
+            const bytecode = [
+                '_H',
+                1,
+                32,
+                'Chrome',
+                32,
+                '$browser',
+                32,
+                'properties',
+                1,
+                2,
+                11,
+                32,
+                '$pageview',
+                32,
+                'event',
+                1,
+                1,
+                11,
+                3,
+                2,
+                4,
+                1,
+            ]
+
+            await createAction(hub.postgres, team.id, 'Test action', bytecode)
+
+            // Create a non-matching event
+            const nonMatchingEvent = createIncomingEvent(team.id, {
+                event: '$pageview',
+                properties: JSON.stringify({ $browser: 'Firefox' }), // Different browser
+                person_id: personId,
+            } as RawClickHouseEvent)
+
+            const filterGlobals = convertClickhouseRawEventToFilterGlobals(nonMatchingEvent)
+            const behavioralEvent: BehavioralEvent = {
+                teamId: team.id,
+                filterGlobals,
+                personId,
+            }
+
+            // Act
+            const result = await (processor as any).processEvent(behavioralEvent)
+
+            // Assert - verify no match occurred
+            expect(result).toBe(0)
+
+            // Assert - check that no counter was written to Cassandra
+            const actions = await hub.actionManagerCDP.getActionsForTeam(team.id)
+            const action = actions[0]
+            const filterHash = createHash('sha256')
+                .update(JSON.stringify(action.bytecode))
+                .digest('hex')
+                .substring(0, 16)
+            const today = new Date().toISOString().split('T')[0]
+
+            const cassandraResult = await cassandra.execute(
+                'SELECT count FROM behavioral_event_counters WHERE team_id = ? AND filter_hash = ? AND person_id = ? AND date = ?',
+                [team.id, filterHash, CassandraTypes.Uuid.fromString(personId), today],
+                { prepare: true }
+            )
+
+            expect(cassandraResult.rows).toHaveLength(0)
+        })
+
+        it('should not write counter when person ID is missing', async () => {
+            // Arrange
+            const bytecode = [
```

Comment:
These bytecodes are all unreadable and in the way of the tests. Maybe an `filters` object with key value to make ti clearer what they are, sort of like what we have for hog function exxamples.